requirements/specifications/intents/user-interest.md

```diff
@@ -0,0 +1,290 @@
+that is communicating with the app, that implies interest of some kind.
+
+**TODO**: This seems like a contractual obligation, as testing this is
+non-trivial / impossible. Are we keeping track of contractual
+requirements anywhere?
+
+## 4. User Interest from a platform UX
+
+Firebolt platforms may provide a platform UX, e.g. voice or and RCU, to
+express user interest in content from an active App. To facilitate this
+Apps will need to be told about the user\'s expressed interest in their
+content.
+
+First, the Aggregated Experience (or some app with the correct
+capability) detects that the user is interested in something. In this
+picture the interest is triggered by an RCU button, but how this occurs
+is outside the scope of this document. When this happens, the Aggregated
+Experience app calls `Content.requestUserInterest()`, which will trigger the
+foreground app's UserInterest provider and call it's `userInterest` method
+by invoking the RPC method `Discovery.onRequestUserInterest`.
+
+![](../../../requirements/images/specifications/intents/user-interest/media/image3.png)
+
+Next, the foreground app receives and responds to the request with an 
+EntityInfo, which is returned as the result to the pending
+`Content.requestUserInterest` method:
+
+![](../../../requirements/images/specifications/intents/user-interest/media/image4.png)
+
+A user's intention to express interest in something is handled by the
+`InterestIntent`. This intent happens *before* the platform knows which
+content is displayed and/or selected by the foreground app. The
+`InterestIntent` may originate from a cloud system, e.g. Voice, or from the RCU, or elsewhere. Origination of this intent is out
+of scope for this document. This intent includes the type of interest,
+e.g. interest or disinterest.
+
+To fire an `InterestIntent` to the current App, another app, typically
+the Aggregated Experience, will call the `Content.requestUserInterest()` method
+of the Discovery SDK.
+
+When `Content.requestUserInterest` is called, if the App in the foreground 
+Lifecycle state has registered a provider for the `xrn:firebolt:capability:discovery:interest` capability, then it **MUST** have that provider invoked with the values from the
+`Content.requestUserInterest` call.
+
+Once an App's callback is invoked, that app will have `interestTimeout`
+milliseconds to return a value or throw an error. Values returned after
+that time **MUST** be ignored. The timeout value is stored in the
+device's configuration manifest.
+
+To be notified when a user expresses interest in the currently displayed
+content, an App **MUST** provide the
+`xrn:firebolt:capability:discovery:interest` capability by enabling the
+`Discovery.onRequestUserInterest` notification.
+
+If there is a valid entity to return, then the method registered by the
+App **MUST** return the currently displayed entity meta-data.
+
+If there is no valid entity to return, then the method **MUST** throw an
```

Comment:
Exception seems harsh here. If the user hits the interest RCU button on something that can't be interested in, seems like that should just another return? Or maybe entity is null?

Reply:
using an exception tells us whether or not to run validation on the result:

if there's a result, validate it, if not, don't.

I'm hoping to move away from JSON-Schemas that support null and some schema.

Reply:
we can make things less harsh in the SDK